File: program_data/github_java_sort_function/7/29.java

    private void shellsort(int[] a) {
        int d = a.length / 2;
        while (true) {
            for (int i = 0; i < a.length; i++) {
                for (int j = i; j + d < a.length; j += d) {
                    int temp;
                    if (a[j] > a[j + d]) {
                        temp = a[j];
                        a[j] = a[j + d];
                        a[j + d] = temp;
                    }
                }
            }
            d--;
            if (d == 1) {
                break;
            }
        }
    }

File: program_data/PKU_raw/69/29.c
//#include <string.h>
char * strrev(char * a){
     char *end;
	 char *beg = a ;
	 char temp;
	 for(end = a; *end; end++);
	 for(end--;beg < end;)
	{
		temp = *beg;
		*beg = *end;
		*end = temp;
		beg++;
        end--;
	}
	return a;
}
int main()
{
	char a[500],b[500] ;
	char *p;
	scanf("%s",a);
	scanf("%s",b);
	
	strrev(a);
	strrev(b);
	
	for(p = a; *p; p++)
		*p = *p - '0';
	for(;p<a+500;p++)
		*p = 0;
	for(p = b; *p; p++)
		*p = *p - '0';
	for(;p<b+500;p++)
		*p = 0;
	int i;
	for(i = 0; i < 500; i++)
	{
		a[i] = a[i] + b[i];
		a[i+1] += a[i]/10;
		a[i] = a[i] % 10;
	}
	
	for(p = a + 400; *p == 0 && p>=a; p--);
	if(p < a)
	printf("0");
	for(;p>=a; p--)
		printf("%d",(int)*p);
	printf("\n");
	//scanf("%s",a);
	return 0; 
}
